### PR TITLE
Make base16 and base32 encodings case preserving

### LINKED ISSUE #18: base16 and base32 encoding should be case preserving
That is is for base 32 if the prefix letter is 'b' the encoding should be lowercase if the prefix letter is 'B' the encoding should be uppercase.  The decoding should be case insensitive.

### PR DESCRIPTION
**Pull request by kevina**
Closes #18

### CODE REVIEW

File: multibase.go [hunk 1]
```diff
@@ -121,3 +133,36 @@ func Decode(data string) (Encoding, []byte, error) {
...
+var base32StdUpperNoPad = base32StdUpperPad.WithPadding(b32.NoPadding)
+
+var base32HexLowerPad = b32.NewEncodingCI("0123456789abcdefghijklmnopqrstuv")
+var base32HexLowerNoPad = base32HexLowerPad.WithPadding(b32.NoPadding)
+
+var base32HexUpperPad = b32.NewEncodingCI("0123456789ABCDEFGHIJKLMNOPQRSTUV")
+var base32HexUpperNoPad = base32HexUpperPad.WithPadding(b32.NoPadding)
+
+
+func hexEncodeToStringUpper(src []byte) string {
```
Comment (Stebalien) [inline multibase.go]: I'm not a fan of embedding all this stuff in this repo but I'm not sure of a better way to do this (without massive over-engineering and making things needlessly complicated).

We should at least move the base32/base16 functions to their own files.

Comment (kevina) [reply multibase.go]: Agreed.  Will do.

Comment (kevina) [reply multibase.go]: And done.

File: base16.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package multibase
+
+import ()
```
Comment (Stebalien) [inline base16.go]: :angry:

Comment (kevina) [reply base16.go]: I take it you want it removed.  I don't see the harm, but don't want to make you any angrier. :smile:

Comment (kevina) [reply base16.go]: Fixed.

**Pull request merged**